sdk/core/core-amqp/src/errors.ts

```diff
@@ -463,6 +463,119 @@ const systemErrorFieldsToCopy: (keyof Omit<NetworkSystemError, "name" | "message
   "syscall"
 ];
 
+/**
+ * AMQP messaging error codes
+ */
+export type MessageErrorCodes =
```

Comment:
Is there any TS magic we can do to ensure that the `ErrorNameConditionMapper` and `ConditionErrorNameMapper` are tied to these codes so that if there are any additions done to one of these and not all three would cause errors?

Reply:
There's always some magic left in the bottle. Will take this on post-this-PR though.

Reply:
Agreed, please log an issue for the backlog

Reply:
Filed: https://github.com/Azure/azure-sdk-for-js/issues/12117